### PR TITLE
working on company auth tokens for users  Issue #147

### PR DESCRIPTION
**Pull request by maedae**

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -41,6 +41,14 @@
     t.datetime "updated_at",    null: false
   end
 
+  create_table "company_invites", force: :cascade do |t|
```
Comment (sumeetjain) [inline db/schema.rb:44]: Always add db index (company_invites => [company_id])

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -22,7 +22,8 @@
       <a href="#">Dashboard</a>
       <%= link_to('Logout', destroy_user_session_path, :method => :delete) %>
     <% else %>
-      <%= link_to('Log In / Create Account', new_user_session_path)  %>  
+      <%= link_to('Sign In', new_user_session_path) %>
+      <%= link_to('Create Account', "/companyInvites/authentication")  %>
```
Comment (sumeetjain) [inline app/views/layouts/application.html.erb:26]: Remove trailing whitespace

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -29,6 +28,7 @@ def configure_permitted_parameters
     devise_parameter_sanitizer.for(:sign_up) << :first_name
     devise_parameter_sanitizer.for(:sign_up) << :last_name
     devise_parameter_sanitizer.for(:sign_up) << :profile_image
+    devise_parameter_sanitizer.for(:sign_up) << :company_id
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb:37]: Calls devise_parameter_sanitizer.for(:sign_up) 5 times (DuplicateMethodCall)

File: app/controllers/companies_controller.rb [hunk 4]
```diff
@@ -6,13 +6,11 @@ def create
   end
 
   def create_confirmation
-    @new_company = Company.new
-    render "edit"
-  end
+    @new_company = Company.new(:profile_image => params[:company][:profile_image], :name => params[:company][:name], :bio => params[:company][:bio], :address_1 => params[:company][:address_1], :address_2 => params[:company][:address_2], :city => params[:company][:city], :state => params[:company][:state], :zip => params[:company][:zip])
```
Comment (sumeetjain) [inline app/controllers/companies_controller.rb:9]: Calls params[:company] 8 times (DuplicateMethodCall)

File: app/controllers/company_invites_controller.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class CompanyInvitesController < ApplicationController
```
Comment (sumeetjain) [inline app/controllers/company_invites_controller.rb:1]: Has no descriptive comment (IrresponsibleModule)

File: app/models/company_invite.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+class CompanyInvite < ActiveRecord::Base
```
Comment (sumeetjain) [inline app/models/company_invite.rb:1]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160324201216_create_company_invites.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class CreateCompanyInvites < ActiveRecord::Migration
+  def change
+    create_table :company_invites do |t|
+      t.integer :company_id
```
Comment (sumeetjain) [inline db/migrate/20160324201216_create_company_invites.rb:4]: Refers to t more than self (maybe move it to another class?) (FeatureEnvy)

File: db/migrate/20160324201216_create_company_invites.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+class CreateCompanyInvites < ActiveRecord::Migration
```
Comment (sumeetjain) [inline db/migrate/20160324201216_create_company_invites.rb:1]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160324201216_create_company_invites.rb [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+  def change
+    create_table :company_invites do |t|
```
Comment (sumeetjain) [inline db/migrate/20160324201216_create_company_invites.rb:3]: Has the variable name 't' (UncommunicativeVariableName)

File: app/controllers/companies_controller.rb (hunk 4 above)
Comment (sumeetjain) [inline app/controllers/companies_controller.rb:9]: Use the new Ruby 1.9 hash syntax.

Comment (sumeetjain) [inline app/controllers/companies_controller.rb:9]: Line is too long. [338/80]

File: app/controllers/companies_controller.rb [hunk 10]
```diff
@@ -6,13 +6,11 @@ def create
...
+    @new_company.save
 
-  def edit
-    @new_company = Company.new(:profile_image => params[:companies][:profile_image], :name => params[:companies][:name], :bio => params[:companies][:bio], :address_1 => params[:companies][:address_1], :address_2 => params[:companies][:address_2], :city => params[:companies][:city], :state => params[:companies][:state], :zip => params[:companies][:zip])
-    render "edit"
+    redirect_to new_user_registration_path(:company_id => @new_company.id)
```
Comment (sumeetjain) [inline app/controllers/companies_controller.rb:12]: Use the new Ruby 1.9 hash syntax.

File: app/controllers/companies_controller.rb [hunk 11]
```diff
@@ -6,13 +6,11 @@ def create
...
   end
 
+
```
Comment (sumeetjain) [inline app/controllers/companies_controller.rb:15]: Extra blank line detected.

Comment (sumeetjain) [inline app/controllers/companies_controller.rb:15]: Extra empty line detected at class body end.

File: app/controllers/company_invites_controller.rb [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
```
Comment (sumeetjain) [inline app/controllers/company_invites_controller.rb:2]: Extra empty line detected at class body beginning.

File: app/controllers/company_invites_controller.rb [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+    render "company_authentication"
+  end
+
+  def company_verification
+    render "company_verification"
+  end
+
+  def company_verification_confirmation
+  end
+
```
Comment (sumeetjain) [inline app/controllers/company_invites_controller.rb:22]: Extra empty line detected at class body end.

File: config/routes.rb [hunk 14]
```diff
@@ -5,6 +5,10 @@
   get '/' => "main#home"
   get '/discover' => "main#discover"
 
+  get 'companyInvites/authentication' => "company_invites#company_authentication"
```
Comment (sumeetjain) [inline config/routes.rb:10]: Line is too long. [81/80]

File: config/routes.rb [hunk 15]
```diff
@@ -5,6 +5,10 @@
...
+  get 'companyInvites/verification' => "company_invites#company_verification"
+  post'companyInvites/verification/confirmation' => "company_invites#company_verification_confirmation"
```
Comment (sumeetjain) [inline config/routes.rb:12]: Line is too long. [103/80]

File: db/schema.rb [hunk 16]
```diff
@@ -11,7 +11,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20160323004716) do
+ActiveRecord::Schema.define(version: 20160324201216) do
```
Comment (sumeetjain) [inline db/schema.rb:14]: Separate every 3 digits in the integer portion of a number with underscores(_).

**Comment by maedae** [comment]
running into this issue with the account creation incorporating company sign-ups. I tried to look up the error and got some information that said the issue could be resolved by clearing browser cookies. However, I still keep running into the issue, so I'm worried that it specifically has something to do with my code. Was wondering if anyone was available to look at it with me?

screen shot 2016-03-24 at 4 12 55 pm

File: app/controllers/application_controller.rb [hunk 17]
```diff
@@ -13,7 +13,12 @@ def store_location
       request.path != "/users/password/new" &&
       request.path != "/users/password/edit" &&
       request.path != "/users/confirmation" &&
-      request.path != "/users/sign_out" &&
+      request.path != "/companies/create/confirmation" &&
+      request.path != "/companies/create" &&
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb:17]: Calls request.path != "/companies/create" 2 times (DuplicateMethodCall)

File: app/controllers/application_controller.rb [hunk 18]
```diff
@@ -13,7 +13,12 @@ def store_location
       request.path != "/users/password/new" &&
       request.path != "/users/password/edit" &&
       request.path != "/users/confirmation" &&
-      request.path != "/users/sign_out" &&
+      request.path != "/companies/create/confirmation" &&
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb:16]: Calls request.path 11 times (DuplicateMethodCall)

**Pull request merged**